### PR TITLE
Splatoon Server

### PR DESCRIPTION
**Pull request by akatsukilevi**
This PR aims to add the Splatoon server to the repository as discussed at the Issue #54.

Currently it makes use of the [ashquarky/splatoon](https://github.com/ashquarky/splatoon) implementation, but once it is merged into Mainline Pretendo, we can replace the submodule URL to match it.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -73,3 +73,7 @@ path = repos/SSSL-DNS
 url = https://github.com/PretendoNetwork/SSSL-DNS.git
 branch = master
 ignore = dirty
+
+[submodule "repos/splatoon"]
+	path = repos/splatoon
```
Comment (MatthewL246) [inline .gitmodules:78]: For consistency, please change the formatting so it matches the rest of the file and add the specified branch and ignore options the others have.

File: compose.yml [hunk 2]
```diff
@@ -459,6 +459,30 @@ services:
...
+    build: ./repos/splatoon
+    depends_on:
+      - account
+      - mongodb
+    restart: unless-stopped
+    ports:
+      # Go delve debugger
+      - 127.0.0.1:2350:2351
+      # Authentication server
+      - 61002:61002/udp
```
Comment (MatthewL246) [inline compose.yml:472]: For consistency with the other servers, this should use ports 6006 and 6007.

File: setup.sh [hunk 3]
```diff
@@ -120,4 +120,4 @@ print_stage "Setting up containers with first-run scripts."
 setup_containers
 
 print_title "Pretendo Network server setup script finished"
-print_success "Setup completed! You can now start your Pretendo Network server with \"docker compose up -d --build\"."
+print_success "Setup completed! You can now start your Pretendo Network server with \"docker compose up -d --build\"."
```
Comment (MatthewL246) [inline setup.sh:123]: Same no newline the end of the file.

File: scripts/setup-environment.sh [hunk 4]
```diff
@@ -115,6 +116,11 @@ echo "PN_WIIU_CHAT_KERBEROS_PASSWORD=$chat_kerberos_password" >>./wiiu-chat.loca
 smm_kerberos_password=$(generate_password 32)
 echo "PN_SMM_KERBEROS_PASSWORD=$smm_kerberos_password" >>./super-mario-maker.local.env
 
+# Generate a Kerberos password for the Splatoon server
+splat_kerberos_password=$(generate_password 32)
+splat_api_key=$(generate_password 32)
```
Comment (MatthewL246) [inline scripts/setup-environment.sh:121]: Looks like an unused variable.

File: environment/splatoon.env [hunk 5]
```diff
@@ -0,0 +1,6 @@
+MONGO_URI=mongodb://mongodb:27017/pretendo_account?replicaSet=rs
+PN_SPLATOON_AUTHENTICATION_SERVER_PORT=61002
```
Comment (MatthewL246) [inline environment/splatoon.env:2]: Same here with the ports.

File: scripts/run-in-container/update-account-servers-database.js [hunk 6]
```diff
@@ -83,6 +83,15 @@ async function runAsync() {
         1,
         "0".repeat(64)
     );
+    await createNexServer(
+        "Splatoon",
+        "10162B00",
+        ["00050000-10176A00", "00050000-10176900", "00050000-10162B00"],
```
Comment (MatthewL246) [inline scripts/run-in-container/update-account-servers-database.js:89]: These title IDs should not include the dash.

File: environment/splatoon.env [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+PN_SPLATOON_SECURE_SERVER_PORT=61003
+PN_SPLATOON_ALLOW_PUBLIC_MATCHMAKING=1
+PN_SPLATOON_ACCOUNT_GRPC_HOST=account
+PN_SPLATOON_ACCOUNT_GRPC_PORT=5000
```
Comment (MatthewL246) [inline environment/splatoon.env:6]: Very minor but this doesn't include a newline at the end of the file, which is a standard I prefer.

**Review by MatthewL246 — DISMISSED** [review]
I just have a few minor formatting and consistency changes. Looks good to me otherwise!

File: scripts/internal/update-account-servers-database.sh [hunk 8]
```diff
@@ -21,6 +21,7 @@ run_verbose docker compose exec -e SERVER_IP="$SERVER_IP" \
     -e MIIVERSE_AES_KEY="$PN_MIIVERSE_API_CONFIG_AES_KEY" \
     -e WIIU_CHAT_PORT="$PN_WIIU_CHAT_AUTHENTICATION_SERVER_PORT" \
     -e SMM_PORT="$PN_SMM_AUTHENTICATION_SERVER_PORT" \
+    -e SPLATOON_PORT="$PN_SPLATOON_AUTHENTICATION_SERVER_PORT" \
```
Comment (MatthewL246) [inline scripts/internal/update-account-servers-database.sh:24]: Caught the issue in CI: this needs to load the Splatoon dotenv files on line 10.

Comment (akatsukilevi) [reply scripts/internal/update-account-servers-database.sh:24]: Was added to load the dotenv file on the right line, shouldn't be causing CI issues now

**Comment by akatsukilevi** [comment]
Testing using Cemu only is being flacky, and don't have 8 Wii U's to test a proper match
Yet, from what I could test so far, things seems to be working as intended

**Comment by MatthewL246** [comment]
Sounds good to me. As far as I can tell, this configuration looks fine.

**Review by MatthewL246 — APPROVED** [review]

**Pull request merged**